cmd/juju/application/upgradecharm.go

```diff
@@ -220,9 +218,6 @@ func (c *upgradeCharmCommand) SetFlags(f *gnuflag.FlagSet) {
 func (c *upgradeCharmCommand) Init(args []string) error {
 	switch len(args) {
 	case 1:
-		if !names.IsValidApplication(args[0]) {
```

Comment:
As described elsewhere, this really is expected to be an application name. Please revert.